src/cpp/src/image_generation/models/unet_inference_static_bs1.hpp

```diff
@@ -0,0 +1,144 @@
+                         const ov::AnyMap& properties) override {
+
+        // All shapes for input/output tensors should be static. 
+        // Double check this and throw runtime error if it's not the case.
+        for (auto& input : model->inputs()) {
+            if (input.get_partial_shape().is_dynamic()) {
+                throw std::runtime_error(
+                    "UNetInferenceStaticBS1::compile: input tensor " + input.get_any_name() +
+                    " shape is dynamic. Tensors must be reshaped to be static before compile is invoked.");
+            }
+        }
+
+        for (auto& output : model->outputs()) {
+            if (output.get_partial_shape().is_dynamic()) {
+                throw std::runtime_error(
+                    "UNetInferenceStaticBS1::compile: output tensor " + output.get_any_name() +
+                    " shape is dynamic. Tensors must be reshaped to be static before compile is invoked.");
+            }
+        }
+
+        // we'll create a separate infer request for each batch.
+        m_nativeBatchSize = model->input("sample").get_shape()[0];
+        m_requests.resize(m_nativeBatchSize);
+
+        //reshape to batch-1
+        UNetInference::reshape(model, 1);
+
+        ov::Core core = utils::singleton_core();
+        ov::CompiledModel compiled_model = core.compile_model(model, device, properties);
+
+        for (int i = 0; i < m_nativeBatchSize; i++ )
+        {
+            m_requests[i] = compiled_model.create_infer_request();
+        }
+    }
+
+    virtual void set_hidden_states(const std::string& tensor_name, ov::Tensor encoder_hidden_states) override {
+        OPENVINO_ASSERT(m_nativeBatchSize && m_nativeBatchSize == m_requests.size(),
+                        "UNet model must be compiled first");
+
+        size_t encoder_hidden_states_bs = encoder_hidden_states.get_shape()[0];
+        if (encoder_hidden_states_bs != m_nativeBatchSize)
+        {
+            throw std::runtime_error("UNetInferenceStaticBS1::set_hidden_states: native batch size is " + std::to_string(m_nativeBatchSize) 
+                + ", but encoder_hidden_states has batch size of " + std::to_string(encoder_hidden_states_bs));
+        }
+
+        char* pHiddenStates = (char *)encoder_hidden_states.data();
+        size_t hidden_states_batch_stride_bytes = encoder_hidden_states.get_strides()[0];
+
+        for (int i = 0; i < m_nativeBatchSize; i++)
+        {
+            auto hidden_states_bs1 = m_requests[i].get_tensor(tensor_name);
+
+            // wrap current pHiddenStates location as batch-1 tensor.
+            ov::Tensor bs1_wrapper(hidden_states_bs1.get_element_type(),
+                                   hidden_states_bs1.get_shape(),
+                                   pHiddenStates,
+                                   encoder_hidden_states.get_strides());
+
+            // copy it to infer request batch-1 tensor
+            bs1_wrapper.copy_to(hidden_states_bs1);
```

Comment:
do you need `copy_to`? 
I think you can just hold a reference to `encoder_hidden_states` in this class to ensure its memory is not destructed. And use `m_requests[i].set_tensor(tensor_name, bs1_wrapper)`

Reply:
So, functionally this would be fine -- and would certainly look more elegant. Unfortunately, it would end up being more memcpy's in the case of NPU. 

The reason is..
When an infer request is created for NPU, the default input / output tensors that are 'attached' to infer_request time at allocated from L0 memory, which is essentially the buffers that will get DMA'ed from. If a user performs a 'set_tensor' with some custom malloc'ed memory, the NPU plugin needs to keep around the DMA-able L0 memory, and will trigger an extra memcpy at each infer() time, copying data from the tensor that user set, to the L0 memory.

So, calling 'set_tensor' will cause an implicit memcpy to be triggered for each infer (i.e. for each unet step), rather than a single memcpy up-front.